src/helpers/bundle-action.type.ts

```diff
@@ -0,0 +1,156 @@
+import { DateString } from '@ukef/helpers';
+
+import { AccrualSchedule } from './accrual-schedule.interface';
+import { RepaymentSchedule } from './repayment-schedule.interface';
+
+export type BundleAction =
+  | AccrualScheduleAmountTransaction
```

Comment:
We don't support all these transaction types. Can we add just the ones that are in Mulesoft, or add 1 by 1 as we need them.

Reply:
The issue with only adding the ones that we use is that the client can give us _any_ bundle identifier so the data we get back from ACBS _might_ be any one of these transaction types. I don't know a better way to represent that accurately in the type system

We initially tried 
```
type BundleAction = NewLoanRequest | { $type: string }
```
but that is too vague for the type system to know we can find out the other fields by checking the `$type`.

By that, I mean that with the way Oscar's written the code, we get
```
if (action.$type === 'NewLoanRequest') {
  // the type system knows `action` matches the `NewLoanRequest` interface
}
```
but with
```
type BundleAction = NewLoanRequest | { $type: string }
```
we lose that:
```
if (action.$type === 'NewLoanRequest') {
  // the type system doesn't know if `action` matches the `NewLoanRequest` interface or the `{ $type: string }` interface
}
```

Reply:
Maybe it is similar to Enums, I assumed that for some fields other values are also possible. Not sure if we can have always up to date list of values in our implementation or we need "other/unknown" values.

Happy to keep longer list of BundleActions if it helps.

Reply:
Used a type predicate to remove the need for the long list, meaning we can just add 1 by 1 as we need.

Reply:
Thanks